core/src/test/scala/unit/kafka/server/ApiVersionsRequestTest.scala

```diff
@@ -17,32 +17,72 @@
 import kafka.test.junit.ClusterTestExtensions
 import org.apache.kafka.server.common.MetadataVersion
 import org.junit.jupiter.api.Assertions._
 import org.junit.jupiter.api.extension.ExtendWith
 
-// TODO: Introduce template in ClusterTests https://issues.apache.org/jira/browse/KAFKA-16595
-//  currently we can't apply template in ClusterTests hence we see bunch of duplicate settings in ClusterTests
+import java.util
+
+object ApiVersionsRequestTest {
+
+  def controlPlaneListenerProperties(): util.HashMap[String, String] = {
+    // Configure control plane listener to make sure we have separate listeners for testing.
+    val serverProperties = new util.HashMap[String, String]()
+    serverProperties.put("control.plane.listener.name", "CONTROL_PLANE")
+    serverProperties.put("listener.security.protocol.map", "CONTROL_PLANE:PLAINTEXT,PLAINTEXT:PLAINTEXT")
+    serverProperties.put("listeners", "PLAINTEXT://localhost:0,CONTROL_PLANE://localhost:0")
+    serverProperties.put("advertised.listeners", "PLAINTEXT://localhost:0,CONTROL_PLANE://localhost:0")
+    serverProperties
+  }
+
+  def testApiVersionsRequestTemplate(clusterGenerator: ClusterGenerator): Unit = {
+    val serverProperties: util.HashMap[String, String] = controlPlaneListenerProperties()
+    serverProperties.put("unstable.api.versions.enable", "false")
+    serverProperties.put("unstable.metadata.versions.enable", "true")
+    clusterGenerator.accept(ClusterConfig.defaultBuilder()
+      .setServerProperties(serverProperties)
+      .setMetadataVersion(MetadataVersion.IBP_3_8_IV0)
+      .build())
+  }
+
+  def testApiVersionsRequestIncludesUnreleasedApisTemplate(clusterGenerator: ClusterGenerator): Unit = {
+    val serverProperties: util.HashMap[String, String] = controlPlaneListenerProperties()
+    serverProperties.put("unstable.api.versions.enable", "true")
+    serverProperties.put("unstable.metadata.versions.enable", "true")
+    clusterGenerator.accept(ClusterConfig.defaultBuilder()
+      .setServerProperties(serverProperties)
+      .build())
+  }
+
+  def testApiVersionsRequestValidationV0Template(clusterGenerator: ClusterGenerator): Unit = {
+    val serverProperties: util.HashMap[String, String] = controlPlaneListenerProperties()
+    serverProperties.put("unstable.api.versions.enable", "false")
+    serverProperties.put("unstable.metadata.versions.enable", "false")
+
+    clusterGenerator.accept(ClusterConfig.defaultBuilder()
+      .setServerProperties(serverProperties)
+      .setMetadataVersion(MetadataVersion.IBP_3_7_IV4)
+      .build())
+  }
+
+  def zkApiVersionsRequest(clusterGenerator: ClusterGenerator): Unit = {
+    clusterGenerator.accept(ClusterConfig.defaultBuilder()
+      .setServerProperties(controlPlaneListenerProperties())
+      .build())
+  }
+}
+
 @ExtendWith(value = Array(classOf[ClusterTestExtensions]))
-@ClusterTestDefaults(brokers = 1)
+@ClusterTestDefaults()
```

Comment:
This can get removed since we will create a default one in generating ClusterConfig